tests/devices/test_mbed_devices.py

```diff
@@ -61,13 +60,6 @@ def test_skips_candidates_without_a_board(self, board, resolve_board, detect_can
             )
         ]
 
-    def test_raises_device_lookup_failed_on_internal_error(self, resolve_board, detect_candidate_devices):
```

Comment:
Shouldn't we test that we bubble up the underlying exception? Maybe this should be checked at a higher layer. I'm thinking we want a test (somewhere) to fail is the exception wasn't bubbled up properly in commit "devices: Remove superfluous exception handling".

Reply:
We don't handle the underlying exception so checking for it here would be a layering issue. We should check in the `targets` module's tests if we raise the correct exceptions from the public interface. I'll add a test in `targets` if a sufficient one doesn't already exist.

Reply:
I've reworked how we handle exceptions so as not to break encapsulation, and to ensure we don't miss any useful logging from target errors. It's in commit https://github.com/ARMmbed/mbed-tools/pull/235/commits/1071175ef4be18ac63f253644df93e121629ad0d 

Please review.